learning/tour-of-beam/backend/internal/fs_content/load.go

```diff
@@ -97,27 +115,57 @@ func collectUnit(infopath string, ctx *sdkContext) (unit *tob.Unit, err error) {
 	return builder.Build(), err
 }
 
+func processTemplate(source []byte, sdk tob.Sdk) ([]byte, error) {
+	t := template.New("")
+	t, err := t.Parse(string(source))
+	if err != nil {
+		return nil, err
+	}
+
+	var output bytes.Buffer
+	err = t.Execute(&output, struct{ Sdk tob.Sdk }{Sdk: sdk})
+	if err != nil {
+		return nil, err
+	}
+
+	return output.Bytes(), nil
+}
+
 func collectGroup(infopath string, ctx *sdkContext) (*tob.Group, error) {
 	info := loadLearningGroupInfo(infopath)
+
+	supported, err := isSupportedSdk(info.Sdk, ctx, infopath)
+	if err != nil {
+		return nil, err
+	}
+	if !supported {
+		log.Printf("Group %v at %v not supported in %v\n", info.Id, infopath, ctx.sdk)
+		return nil, nil
+	}
+
 	log.Printf("Found Group %v metadata at %v\n", info.Name, infopath)
 	group := tob.Group{Id: info.Id, Title: info.Name}
 	for _, item := range info.Content {
 		node, err := collectNode(filepath.Join(infopath, "..", item), ctx)
 		if err != nil {
 			return &group, err
 		}
-		group.Nodes = append(group.Nodes, node)
+		if node == nil {
+			continue
+		}
+		group.Nodes = append(group.Nodes, *node)
 	}
 
 	return &group, nil
 }
 
 // Collect node which is either a unit or a group.
-func collectNode(rootpath string, ctx *sdkContext) (node tob.Node, err error) {
+func collectNode(rootpath string, ctx *sdkContext) (*tob.Node, error) {
 	files, err := os.ReadDir(rootpath)
 	if err != nil {
-		return node, err
+		return nil, err
 	}
+	node := &tob.Node{}
 	for _, f := range files {
```

Comment:
This isn't technically a problem with this code change, but I think this loop has a potential problem. If the first file has an associated error, but the second does not it will reset `err` to `nil`

Probably we should be early returning if `err != nil` after the switch

Reply:
Thank you! Nice catch!